### ISSUE TITLE
Is it possible to test distrubuted file systems like HDFS using the existed models? 

### ISSUE THREAD

**Issue by hexiecs** [jepsen-io/knossos#32]
Well, I want to test the linearizability of my own distributed file system, which is similar to HDFS. And I have seen there are some useful models already such as cas-register and multi-register, but I don't know how to apply these models to the file systems, which is a little different from a key-value store. For example, in a file system, writing/reading is at a certain offset while the key-value store is writing/reading the key's whole value. I want to ask if there are any possibilities that I can use the existed models here or I have to write my own models?
Since I'm not an English native speaker, my sentences or words above may not accurate, sorry in advance and looking forward to your reply!

**Comment by aphyr** [comment]
Yeah, I think you'd need to build a different kind of model that represents the sorts of operations you'd like to perform on your filesystem. I'm not sure what that would look like!
